packages/immutadot-benchmark/src/updateTodos.spec.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-env jest */
+
+import { List, Record } from 'immutable'
+
+import immer, { setAutoFreeze, setUseProxies } from 'immer'
+
+import { createBenchmark } from './benchmark'
+
+import { set } from 'immutadot/core'
+
+function updateTodosList(title, listSize, modifySize, maxTime, maxOperations) {
+  // Prepare base state
+  const baseState = []
+  for (let i = 0; i < listSize; i++) {
+    baseState.push({
+      todo: `todo_${i}`,
+      done: false,
+      someThingCompletelyIrrelevant: [1, 2, 3, 4, 5, 6, 7, 8, 9, 0],
+    })
+  }
+
+  // Prepare immutalbe state
+  const todoRecord = Record({
+    todo: '',
+    done: false,
+    someThingCompletelyIrrelevant: [],
+  })
+  const immutableState = List(baseState.map(todo => todoRecord(todo)))
+
+  // Disable immer auto freeze
+  setAutoFreeze(false)
+
+  const unmodifiedSize = listSize - modifySize
+  const randomStart = () => Math.floor(Math.random() * unmodifiedSize)
+
+  const benchmark = createBenchmark(
+    title,
+    (key, result) => {
+      if (key === 'immutable') return
+      let trues = 0, falses = 0
+      result.forEach(todo => todo.done ? trues++ : falses++)
+      expect(trues).toBe(modifySize)
+      expect(falses).toBe(unmodifiedSize)
+    },
+    maxTime,
+    maxOperations,
+  )
+
+  it('ES2015', () => {
+    benchmark('es2015', 'ES2015 destructuring', () => {
+      const start = randomStart(), end = start + modifySize
```

Comment:
It's more readable than the previous one with ternary operators but i found it hard to view each variables at one glance.